src/nvim/os/fs.c

```diff
@@ -460,8 +460,13 @@ ptrdiff_t os_read(const int fd, bool *ret_eof, char *const ret_buf,
   bool did_try_to_free = false;
   while (read_bytes != size) {
     assert(size >= read_bytes);
+#ifdef WIN32
+    const ptrdiff_t cur_read_bytes = read(fd, ret_buf + read_bytes,
+                                          (unsigned int)(size - read_bytes));
```

Comment:
Better conditionally define a `LEN_CAST` macros above the function (and undefine it below the last function code is being used in). (Local defining/undefining like this allows using shorter names for local macros without fear of running into collisions.) And still do not use `int`, `unsigned int` and `unsigned` are the same thing.